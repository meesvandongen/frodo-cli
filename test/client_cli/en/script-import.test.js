import cp from 'child_process';
import { promisify } from 'util';

const exec = promisify(cp.exec);
const CMD = 'frodo script import --help';
const { stdout } = await exec(CMD);

test("CLI help interface for 'script import' should be expected english", async () => {
  expect(stdout).toMatchSnapshot();
});
