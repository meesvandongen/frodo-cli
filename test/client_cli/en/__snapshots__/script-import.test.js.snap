// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI help interface for 'script import' should be expected english 1`] = `
"Usage: frodo script import [options] <host> [realm] [user] [password]

Import scripts.

Arguments:
  host                      Access Management base URL, e.g.:
                            https://cdk.iam.example.com/am. To use a connection
                            profile, just specify a unique substring.
  realm                     Realm. Specify realm as '/' for the root realm or
                            'realm' or '/parent/child' otherwise. (default:
                            "alpha" for Identity Cloud tenants, "/" otherwise.)
  user                      Username to login with. Must be an admin user with
                            appropriate rights to manage authentication
                            journeys/trees.
  password                  Password.

Options:
  -m, --type <type>         Override auto-detected deployment type. Valid
                            values for type:
                            classic:  A classic Access Management-only
                            deployment with custom layout and configuration.
                            cloud:    A ForgeRock Identity Cloud environment.
                            forgeops: A ForgeOps CDK or CDM deployment.
                            The detected or provided deployment type controls
                            certain behavior like obtaining an Identity
                            Management admin token or not and whether to
                            export/import referenced email templates or how to
                            walk through the tenant admin login flow of
                            Identity Cloud and handle MFA (choices: "classic",
                            "cloud", "forgeops")
  -k, --insecure            Allow insecure connections when using SSL/TLS. Has
                            no effect when using a network proxy for https
                            (HTTPS_PROXY=http://<host>:<port>), in that case
                            the proxy must provide this capability. (default:
                            Don't allow insecure connections)
  --verbose                 Verbose output during command execution. If
                            specified, may or may not produce additional
                            output.
  --debug                   Debug output during command execution. If
                            specified, may or may not produce additional output
                            helpful for troubleshooting.
  --curlirize               Output all network calls in curl format.
  -f, --file <file>         Name of the file to import.
  -n, --script-name <name>  Name of the script. If specified, -a and -A are
                            ignored.
  --re-uuid                 Re-UUID. Create a new UUID for the script upon
                            import. Use this to duplicate a script or create a
                            new version of the same script. Note that you must
                            also choose a new name using -n/--script-name to
                            avoid import errors. (default: false)
  -s, --script <script>     DEPRECATED! Use -n/--script-name instead. Name of
                            the script.
  -h, --help                Help
"
`;
